cloudinit/config/cc_chef.py

```diff
@@ -298,23 +300,27 @@ def subp_blob_in_tempfile(blob, *args, **kwargs):
     """
     basename = kwargs.pop("basename", "subp_blob")
 
-    if len(args) == 0 and "args" not in kwargs:
-        args = [tuple()]
+    _args = list(args)
+    if len(_args) == 0 and "args" not in kwargs:
+        _args.append(tuple())
```

Comment:
If this branch is taken then on line 315 we will be concatenating a list and a tuple, which will cause a typerror.
```python
_args[0] = [tmpf] + _args[0]
```